warrior/Framework/Utils/data_Utils.py

```diff
@@ -900,15 +897,29 @@ def validate():
     result, err_msg, exp = validate()
     keys = key.split('.')
     value = get_object_from_datarepository(keys[0])
-    for k in keys[1:]:
-        value = value[k]
-    if not value:
-        err_msg += "key {} not present in data repository\n".format(key)
+    key_err_msg = "key {} not present in data repository\n"
+    if value:
+        try:
+            for k in keys[1:]:
+                value = value[k]
+            if result == "ERROR" or result == "EXCEPTION":
+                print_error(err_msg)
+            elif not comp_funcs[comparison](value, exp):
+                result = "FALSE"
+                print_warning("The key, value pair '{0}:{1}' present in the  "
+                              "data_repository doesn't satisfy the expected value & "
+                              "condition '{2}:{3}'".format(key, value, comparison, expected))
```

Comment:
Hi @sugan19,

In line 911, ```format(key, value, comparison, expected)``` would print out the ```key``` argument rather than ```k``` from the for loop in line 903 - was this intentional?

Thanks.

Reply:
@sanikakulkarni Yes, this was intentional. The 'key' value has the entire key (eg: if user gives k1.k2.k3) but k has only k1/k2/k3.

Reply:
Hi @sugan19,

Since this print statement prints out a negative scenario, it would be best if the entire key ( key ) is printed out as well as specifically ( k ) is also printed out.

Consider the scenarios:

1. In the current print statement, the user has no way to know was missing (since k is not being printed separately)
2. If you just print out "k", user may not be able to figure out which "k" is missing as "key" may have duplicate "k"s

Hence, printing "key" and "k" would be the most beneficial

Thanks.

Reply:
Hi @sugan19,

I am not sure if I understand what you mean by ```The print_warning in line: 909 is executed only if it has the entire key and k has no scope in elif block.```. Can you please elaborate?

Thanks.